Firestore/core/src/firebase/firestore/core/view_snapshot.h

```diff
@@ -179,6 +176,9 @@ class ViewSnapshot {
   bool excludes_metadata_changes_ = false;
 };
 
+using ViewSnapshotListener = std::unique_ptr<EventListener<ViewSnapshot>>;
```

Comment:
Question: how does moving these aliases out of the class help?

Reply:
Nested classes cannot be forward declared. That is, this isn't legal:

```c++
class ViewSnapshot;
using ViewSnapshot::Listener = std::unique_ptr<EventListener<ViewSnapshot>>;
```

However, this is legal:

```c++
class ViewSnapshot;
using ViewSnapshotListener = std::unique_ptr<EventListener<ViewSnapshot>>;
```